### PR TITLE
DOC-1949 - Config API spec changes

### PR DESCRIPTION
**Pull request by jcullen-toast**
Made changes to the contact, added host, added operationIds, removed [a href]s, reduced top-level description, reduced most summaries.

I left a couple summaries related to modifier groups as is, but included suggestions as comments. Not sure what to do about the tags.

### CODE REVIEW

File: openapi/toast-config-api.yaml [hunk 1]
```diff
@@ -955,12 +930,11 @@ paths:
             $ref: '#/definitions/AlternatePaymentType'
   /alternatePaymentTypes:
     get:
+      operationId: alternatePaymentTypesGet
       summary: |
-        Returns an array of alternative forms of payment configured for
-        a restaurant.
+        Get details on alternative payment types
```
Comment (lizjackson-toast) [inline openapi/toast-config-api.yaml:935]: Nit: add period at end of sentence

File: openapi/toast-config-api.yaml [hunk 2]
```diff
@@ -1013,10 +987,11 @@ paths:
               $ref: '#/definitions/AlternatePaymentType'
   /breakTypes/{guid}:
       get:
+        operationId: breakTypesGuidGet
         summary: |
-          Returns information about a break configured for a restaurant.
+          Get details on a break
```
Comment (lizjackson-toast) [inline openapi/toast-config-api.yaml:992]: How about `Returns information about a break type configured for a restaurant.`? I say this because the endpoint returns information about break _types_ and not about breaks themselves (break types are static, breaks are transactional)

File: openapi/toast-config-api.yaml [hunk 3]
```diff
@@ -1013,10 +987,11 @@ paths:
...
         description: |
-          Returns a <a href="#/definitions/BreakType">`BreakType`</a>
+          Returns a `BreakType`
           object containing information about a break configured for a
           restaurant.
```
Comment (lizjackson-toast) [inline openapi/toast-config-api.yaml:996]: About a break type (not about a break)

File: openapi/toast-config-api.yaml [hunk 4]
```diff
@@ -1040,11 +1015,11 @@ paths:
               $ref: '#/definitions/BreakType'
   /breakTypes:
       get:
+        operationId: breakTypesGet
         summary: |
-          Returns an array of breaks configured for a restaurant.
+          Get details on all breaks
```
Comment (lizjackson-toast) [inline openapi/toast-config-api.yaml:1020]: `Returns an array of break types configured for a restaurant.` ?

File: openapi/toast-config-api.yaml [hunk 5]
```diff
@@ -1092,10 +1067,11 @@ paths:
                 $ref: '#/definitions/BreakType'
   /cashDrawers/{guid}:
     get:
+      operationId: cashDrawersGuidGet
       summary: |
-        Returns information about a specific cash drawer device.
+        Get details on a cash drawer
```
Comment (lizjackson-toast) [inline openapi/toast-config-api.yaml:1072]: How about `Returns information about a specific cash drawer device.` "Returns" seems more specific than "Gets", and I think of "returns...about" as more idiomatic than "gets...on". How does that sound?

File: openapi/toast-config-api.yaml (hunk 1 above)
Comment (jcullen-toast) [reply openapi/toast-config-api.yaml:935]: Doesn't the new format exclude periods at the end?

File: openapi/toast-config-api.yaml (hunk 2 above)
Comment (jcullen-toast) [reply openapi/toast-config-api.yaml:992]: @lizjackson-toast How about "Get information about a break type"?

File: openapi/toast-config-api.yaml (hunk 4 above)
Comment (jcullen-toast) [reply openapi/toast-config-api.yaml:1020]: @lizjackson-toast How about 'Get information about break types'?

File: openapi/toast-config-api.yaml (hunk 5 above)
Comment (jcullen-toast) [reply openapi/toast-config-api.yaml:1072]: 'Returns information about a specific cash drawer device.' is what was there originally, so I'm trying to shorten that. How about 'Get information about a cash drawer'?

Comment (lizjackson-toast) [reply openapi/toast-config-api.yaml:1072]: Mentioned this to you in Slack, but this is my bad. I was thinking of these sentences as the description and not the summary. Long day. "Get a cash drawer" should work.

File: openapi/toast-config-api.yaml [hunk 6]
```diff
@@ -921,12 +897,11 @@ definitions:
 paths:
   /alternatePaymentTypes/{guid}:
     get:
+      operationId: alternatePaymentTypesGuidGet
       summary: |
-        Returns information about an alternative form of payment
-        configured for a restaurant.
+        Get details on an alternative payment type
```
Comment (pdesjardins) [inline openapi/toast-config-api.yaml]: I prefer "get information about..."

How do you feel about replacing "details on"? It doesn't feel right to me.

File: openapi/toast-config-api.yaml (hunk 1 above)
Comment (pdesjardins) [reply openapi/toast-config-api.yaml]: I think you're right. These summaries don't include periods.

Comment (pdesjardins) [reply openapi/toast-config-api.yaml]: Replace "details on" with "information about" globally in the file, I think.

File: openapi/toast-config-api.yaml (hunk 5 above)
Comment (pdesjardins) [reply openapi/toast-config-api.yaml]: `Get information about a cash drawer`

???

File: openapi/toast-config-api.yaml [hunk 7]
```diff
@@ -1504,13 +1528,15 @@ paths:
               $ref: '#/definitions/MenuItem'
   /menuOptionGroups/{guid}:
     get:
+      tags: 
+        - Menu Option Groups
```
Comment (pdesjardins) [inline openapi/toast-config-api.yaml:1532]: Should the tag name be `Modifier Groups`?

I thought the new and preferred terminology was "modifier group" instead of "modifier option group." "Menu group" sounds like the really early terminology that was baked into the codebase in 2015. Even though we can't change the endpoint name, we can use an updated tag name, I think.

File: openapi/toast-config-api.yaml [hunk 8]
```diff
@@ -4,31 +4,46 @@ info:
...
   - application/json
+tags: 
+  - name: Alternate Payment Types
+  - name: Break Types
+  - name: Cash Drawers
+  - name: Dining Options
+  - name: Discounts
+  - name: Menu Groups
+  - name: Menu Items
+  - name: Menu Option Groups
```
Comment (pdesjardins) [inline openapi/toast-config-api.yaml:30]: Should this be `Modifier Groups` to use the latest preferred terminology?

**Review by pdesjardins — APPROVED** [review]
I left a few comments. This is a big improvement in this really big spec. 👍

File: openapi/toast-config-api.yaml (hunk 6 above)
Comment (jcullen-toast) [reply openapi/toast-config-api.yaml]: @pdesjardins Would you prefer 'Get information about...' instead of 'Get...'? For example, "Get an alternative payment type".

Comment (pdesjardins) [reply openapi/toast-config-api.yaml]: Well, I do prefer "get information about an alternative" over "get an alternative." Even though it has more words. "Get an alternative payment type" feels like developer shorthand, and I prefer to spell out what the endpoint does in plain English.

But, I'm OK with "get an alternative payment type." Especially so that we can get the overall updates you're making into the site.

Comment (lizjackson-toast) [reply openapi/toast-config-api.yaml]: I like "Get an alternative payment type", "Get a cash drawer", etc. "Get information about..." feels redundant; it's implied that they are getting information.

I'm thinking of the Square docs for reference:

image

Also Spreedly:

image

And Clover:

image

What do you think?

Comment (pdesjardins) [reply openapi/toast-config-api.yaml]: OK! I'm convinced. "Information about" is implied and understood in the API context.

Our endpoints should "get an alternative payment type."

Comment (jcullen-toast) [reply openapi/toast-config-api.yaml]: In that case, I'm going to merge. :)
